docs/site/content/docs/latest/storage.md

```diff
@@ -0,0 +1,146 @@
+
+The Tanzu default storage class definitions are:
+
+### vSphere CNS
+
+```sh
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: default
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: csi.vsphere.vmware.com
+parameters:
+  storagePolicyName: optional
+```
+
+See the [vSphere](https://kubernetes.io/docs/concepts/storage/storage-classes/#vsphere) CSI storage class parameters in the Kubernetes documentation.
+
+### Amazon EBS
+
+```sh
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: default
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
+provisioner: kubernetes.io/aws-ebs
+```
+
+See the [Amazon EBS](https://kubernetes.io/docs/concepts/storage/storage-classes/#aws-ebs) storage class parameters in the Kubernetes documentation.
+
+### Azure Disk
+
+```sh
+apiVersion: storage.k8s.io/v1beta1
+kind: StorageClass
+metadata:
+  name: default
+  annotations:
+    storageclass.beta.kubernetes.io/is-default-class: "true"
+  labels:
+    kubernetes.io/cluster-service: "true"
+provisioner: kubernetes.io/azure-disk
+parameters:
+  kind: Managed
+  storageaccounttype: Standard_LRS
+  cachingmode: ReadOnly
+volumeBindingMode: WaitForFirstConsumer
+```
+
+See the [Azure Disk](https://kubernetes.io/docs/concepts/storage/storage-classes/#aws-ebs) storage class parameters in the Kubernetes documentation.
+
+## Create a Custom Storage Class
+
+Cluster administrators can create a new storage class as follows:
+
+Before You Begin:
+
+(vSphere only) Select or create the VM storage policy to use as the basis for the Kubernetes `StorageClass`. vSphere administrators can create a storage policy by following the steps in [Set Up vSphere CNS and Create a Storage Policy in vSphere](vsphere-cns).
+
+1. Create a `StorageClass` configuration `.yaml` with `provisioner`, `parameters`, and other options.
+2. (vSphere only) Associate a Kubernetes storage class with a vSphere storage policy by setting its `storagePolicyName` parameter to the vSphere storage policy name, as a double-quoted string.
+3. Pass the file to `kubectl create -f`
+4. Verify the storage class by running `kubectl describe storageclass <storageclass metadata.name>`.
+
+For example, see [Enabling Dynamic Provisioning](https://kubernetes.io/docs/concepts/storage/dynamic-provisioning/#enabling-dynamic-provisioning) in the Kubernetes documentation.
+
+For vSphere CSI information and resources, see [VMware vSphere Container Storage Plug-in Documentation](https://docs.vmware.com/en/VMware-vSphere-Container-Storage-Plug-in/index.html).
+
+## Use a Custom Storage Class in a Cluster
+
+To provision persistent storage for their cluster nodes that does not use one of the [Default Storage Classes](#defaults) described above, cluster users include a custom storage class in a pod configuration as follows:
+
+1. Set the context of `kubectl` to the cluster. For example:
+
+  ```sh
```

Comment:
This code block isn't indented enough, so it isn't getting formatted correctly in the rendered output. Just needs an additional space before each line to line up with the "S" in the top bullet item "Set".